etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2476 @@
+*
+* @example
+* // Bad...
+* class MyClass {
+*   memberWithoutVisibility = 'no explicit visibility';
+* }
+*
+* // Good...
+* class MyClass {
+*   private memberWithVisibility = 'has explicit visibility';
+* }
+*/
+rules[ '@typescript-eslint/explicit-member-accessibility' ] = [
+	'error',
+	{
+		'accessibility': 'no-public'
+	}
+];
+
+/**
+* Requires explicit return and argument types on exported functions' and classes' public class methods.
+*
+* @name explicit-module-boundary-types
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [explicit-module-boundary-types]{@link https://typescript-eslint.io/rules/explicit-module-boundary-types}
+*
+* @example
+* // Bad...
+* export function badFunction(a) {
+*   return a;
+* }
+*
+* // Good...
+* export function goodFunction(a: string): string {
+*   return a;
+* }
+*/
+rules[ '@typescript-eslint/explicit-module-boundary-types' ] = 'error';
+
+/**
+* Enforces indentation with tabs.
+*
+* @name indent
+* @memberof rules
+* @type {Array}
+* @default [ 'error', 'tab' ]
+* @see [indent]{@link https://typescript-eslint.io/rules/indent}
+*
+* @example
+* // Bad...
+* function bad() {
+*  return true;
+* }
+*
+* // Good...
+* function good() {
+*   return true;
+* }
+*/
+rules[ '@typescript-eslint/indent' ] = [ 'error', 'tab' ];
+
+/**
+* Enforces a specific member delimiter style in interfaces and type literals.
+*
+* @name member-delimiter-style
+* @memberof rules
+* @type {Array}
+* @default [ 'error', { 'multiline': { 'delimiter': 'semi', 'requireLast': true }, 'singleline': { 'delimiter': 'semi', 'requireLast': false } }]
+* @see [member-delimiter-style]{@link https://typescript-eslint.io/rules/member-delimiter-style}
+*
+* @example
+* // Bad...
+* interface Bad {
+*   name: string,
+*   age: number
+* }
+*
+* // Good...
+* interface Good {
+*   name: string;
+*   age: number;
+* }
+*/
+rules[ '@typescript-eslint/member-delimiter-style' ] = [
+	'error',
+	{
+		'multiline': {
+			'delimiter': 'semi',
+			'requireLast': true
+		},
+		'singleline': {
+			'delimiter': 'semi',
+			'requireLast': false
+		}
+	}
+];
+
+/**
+* Enforces naming conventions for everything across a codebase.
+*
+* @name naming-convention
+* @memberof rules
+* @type {Array}
+* @default [ 'error', { 'selector': 'variable', 'format': [ 'camelCase', 'UPPER_CASE' ], 'leadingUnderscore': 'forbid', 'trailingUnderscore': 'forbid' }]
+* @see [naming-convention]{@link https://typescript-eslint.io/rules/naming-convention}
+*
+* @example
+* // Bad...
+* const BAD_NAME = 1;
+* const bad_Name = 2;
+*
+* // Good...
+* const goodName = 3;
+* const GOOD_NAME = 4;
+*/
+rules[ '@typescript-eslint/naming-convention' ] = [
+	'error',
+	{
+		'selector': 'variable',
+		'format': [ 'camelCase', 'UPPER_CASE' ],
+		'leadingUnderscore': 'forbid',
+		'trailingUnderscore': 'forbid'
+	}
+];
+
+/**
+* Disables TypeScript's no-shadow rule.
+*
+* @name no-shadow
+* @memberof rules
+* @type {Array}
+* @default [ 'off', { 'hoist': 'all' }]
```

Comment:
```suggestion
* @default [ 'off', { 'hoist': 'all' } ]
```

Reply:
@Planeshifter How does this rule work? For `off`, I would have just expected the literal as the rule value, but instead we also include configuration here (e.g., `{ 'hoist': 'all' }`).